src/StudyViewer/ExportToWorkspace.jsx

```diff
@@ -0,0 +1,169 @@
+import { fetchWithCreds } from '../actions';
+
+class ExportToWorkspace extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toasterOpen: false,
+      toasterHeadline: '',
+      exportPFBToWorkspaceGUID: '',
+      exportPFBToWorkspaceStatus: null,
+      jobFailed: false,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.exportToWorkspaceAction
+      && this.props.exportToWorkspaceAction.accessibleValidationValue
+      && !this.props.exportingPFBToWorkspace
+      && this.props.exportToWorkspaceAction.accessibleValidationValue !== prevProps.exportToWorkspaceAction.accessibleValidationValue) {
+      this.exportToWorkspace(this.props.exportToWorkspaceAction);
+    }
+    if (this.props.job && !this.props.job.status && !this.state.jobFailed) {
+      this.onJobFailed('There was an error please refresh the page and try again');
+    }
+    if (this.props.job && this.props.job.status === 'Failed' && prevProps.job && prevProps.job.status !== 'Failed' && !this.state.jobFailed) {
+      this.onJobFailed('There was an error exporting your cohort.');
+    }
+    if (this.props.job && this.props.job.status === 'Completed' && prevProps.job && prevProps.job.status !== 'Completed') {
+      this.fetchJobResult()
+        .then((res) => {
+          if (this.props.exportingPFBToWorkspace) {
+            const pfbGUID = `${res.data.output}`.split('\n')[0];
+            this.sendPFBToWorkspace(pfbGUID);
+          }
+        });
+    }
+  }
+
+  onJobFailed = (toasterHeadline) => {
+    this.setState({
+      toasterOpen: true,
+      toasterHeadline,
+      jobFailed: true,
+    });
+    this.props.exportingPFBToWorkspaceStateChange(false);
+  }
+
+  exportToWorkspace = (buttonConfig) => {
+    this.props.exportingPFBToWorkspaceStateChange(true);
+    this.setState({
+      toasterOpen: true,
+      exportPFBToWorkspaceGUID: '',
+      jobFailed: false,
+    });
+
+    this.props.submitJob({
+      action: 'export',
+      access_format: 'guid',
+      input: { filter: { '=': { auth_resource_path: buttonConfig.accessibleValidationValue } }, root_node: buttonConfig.root_node },
+    });
+    this.props.checkJobStatus();
+    this.setState({
+      toasterHeadline: 'Your export is in progress.',
+    });
+  };
+
+  sendPFBToWorkspace = (pfbGUID) => {
+    const JSONBody = { guid: pfbGUID };
+    fetchWithCreds({
+      path: `${manifestServiceApiPath}cohorts`,
+      body: JSON.stringify(JSONBody),
+      method: 'POST',
+    })
+      .then(
+        ({ status, data }) => {
+          const errorMsg = (data.error ? data.error : '');
+          this.props.exportingPFBToWorkspaceStateChange(false);
+          switch (status) {
+          case 200:
+            this.setState(() => ({
+              exportPFBToWorkspaceGUID: pfbGUID,
+              toasterOpen: true,
+              toasterHeadline: 'A PFB for this cohort will be saved to your workspace. The GUID for your PFB is displayed below.',
```

Comment:
i know this is what's displayed for the same functionality in the explorer, but here i think we'll want to abstract the technical details - the sponsors may look at this closely and be confused by "GUID" and "PFB"; "cohort" also doesn't make sense for NIAID. Could we simplify the message to "The data has been exported to your workspace"?